project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java

```diff
@@ -1531,6 +1531,63 @@ public void testAutoUpdateWithBuildFile() throws IOException {
         Assert.assertFalse(thirdBuildJson.isExpiredLastUpdateTime());
     }
 
+    @Test(description = "test auto updating dependencies with build file after removing Dependencies.toml")
+    public void testAutoUpdateWithBuildFileWithoutDepsToml() throws IOException {
+        Path projectPath = RESOURCE_DIRECTORY.resolve("myproject");
+        // Delete build file and Dependencies.toml file if already exists
+        Files.deleteIfExists(projectPath.resolve(TARGET_DIR_NAME).resolve(BUILD_FILE));
+        Files.deleteIfExists(projectPath.resolve(DEPENDENCIES_TOML));
+
+        // Set sticky false, to imitate the default build command behavior
+        BuildOptionsBuilder buildOptionsBuilder = new BuildOptionsBuilder();
+        buildOptionsBuilder.sticky(false);
```

Comment:
When we delete the build file and dependencies.toml, do we need to explicitly set `sticky` false?

Reply:
Removed using these build options for the first build, but anyway we have to use it for the second build.